database/sqlite3/sqlite3.go

```diff
@@ -268,3 +267,14 @@ func (m *Sqlite) Version() (version int, dirty bool, err error) {
 	}
 	return version, dirty, nil
 }
+
+func dbPathFromURL(url *nurl.URL) string {
+	dbPath := url.Path
+	encodedQuery := migrate.FilterCustomQuery(url).Query().Encode()
+
+	if len(encodedQuery) > 0 {
+		dbPath = dbPath + "?" + encodedQuery
```

Comment:
Use `+=` or return directly. If you return directly, it's cleaner to invert your conditional.

Reply:
Flipped the logic to return `dbPath` if len == 0, otherwise returns the `dbPath + "?" + encodedQuery` statement directly.